### PR TITLE
Kraken: Add RT route dynamically

### PR DESCRIPTION
**Pull request by pbougue**
For history on comments, this PR is exactly this draft: https://github.com/pbougue/navitia/pull/2
A lot of discussions were addressed there.
JIRA: https://jira.kisio.org/browse/NAVP-1064

:mag_right: please review by commit (and read commit message).
Only 2 commits are "canceling one-another", but it's highlighting the way things work (commits 2 & 3, `fix tests` and `remove fix`).

TODO (only what's left from draft):
- [x] Add kraken realtime_tests on objects created to check that there are no `nullptr` left.

> Another PR follows, using news methods in build_helper (to challenge methods more): #2729

### CODE REVIEW

**Review by benoit-bst — COMMENTED** [review]
I didn't read the python tests part, but for the rest I'm okay :tada: 

Just for the **QA**, can you test the _autocomplete part_, with a playground ? 
After an added trip, I would like to know if you have the expected result.

> Done in integration test (`/pt_objects` before and after): https://github.com/CanalTP/navitia/pull/2724/files#diff-ca32dae0e1de944fd72a73775b54cfe1R1614

**Comment by pbougue** [comment]
Jenkins test this please

**Comment by pbougue** [comment]
Jenkins, test this please.

**Review by kadhikari — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]

File: source/kraken/apply_disruption.cpp [hunk 1]
```diff
@@ -182,6 +182,21 @@ static std::string concatenate_impact_uris(const nt::MetaVehicleJourney& mvj) {
     return impacts_uris.str();
 }
 
+
+static nt::Route* get_or_create_route(const nt::disruption::Impact& impact, nt::PT_Data& pt_data) {
+    nt::Network* network = pt_data.get_or_create_network("network:additional_service", "additional service");
+    nt::CommercialMode* comm_mode = pt_data.get_or_create_commercial_mode("commercial_mode:additional_service",
+                                                                          "additional service");
+    // TODO: manage line.code when necessary
```
Comment (pbougue) [inline source/kraken/apply_disruption.cpp:190]: This should not be necessary for SNCF data at least (empty in base_schedule data)

**Pull request merged**